### ISSUE TITLE
[OTG/BiomeBundle] option to grow anywhere

### ISSUE THREAD

**Issue by ghost** [MatrexsVigil/harvestcraft#320]
First off, don't automatically shoot this down because the other mod is causing the problem.

Would it be possible to add an option to change the biome whitelist into a blacklist? This would save a lot of time because I could simply add swamps and ice biomes to a blacklist rather than bloat the config file to 75x its size adding everything from biomebundle and all the other biomes.

**Comment by fllo** [comment]
I'd prefer some good defaults, not a blacklist.
I will look into creating great defaults for OTG/BiomeBundle, Biomes'O'Pleny and vanillia.
It will take some time.
The following is the data that will be used:
Edit: Moved to gist
https://gist.github.com/fllo/62e443c04b4fdf383bb3f90f79f83a19

**Comment by ghost** [comment]
While this does allow for fine tuning, and I can appreciate that, this presents the aforementioned problem of a massive config file. You have to keep in mind that for every tree you have to insert the appropriate biomes, since they aren't universal. Given that my pack also has serene seasons installed, the effect of a tree a tad out of place is diminished. My goal in the suggestion was to create the least amount of work for both me and any contributors for what is very likely an extremely uncommon configuration.

Additionally, given the fact that I have serene seasons installed, I cannot have anything else messing with rain and snow. I also have other mods installed that have forced me to modify biome bundle a bit, so any edits to that mods config will be overridden.

**Comment by fllo** [comment]
I will look into serene seasons.
I would also move away from using the biome names, as it is currently being used. And use the properties like temperature, canRain etc for selection of fruit trees to spawn. So it will be more generic.

**Comment by ghost** [comment]
That makes more sense, and would also mean that it could be set up to essentially work with any terrain generation mod that has biome temperatures (which is most all of them AFAIK).

Also, I think I misread that giant table as meaning setting the biome conditions, rather than reading them, my bad >.>

**Comment by fllo** [comment]
It would be great if "no manual configuration" is needed, and it just works.

The large table will be used for finding the best possbile way, without the need of a white/blacklist.

**Comment by ghost** [comment]
Agreed, things that just work without needing prodded are much appreciated when you have over 250 mods.

**Comment by fllo** [comment]
OTG BiomeBundle, seems to use for its BiomeHeight mainly negative numbers. And then uses those for baseHeight.
It doesn't make any sense. http://openterraingen.wikia.com/wiki/Biome_Configs

```
Forge 1.12.2-14.23.5.2768 in Biome.java:
76 /** The base height of this biome. Default 0.1. */
77    private final float baseHeight;
```

reg name | Name | Temperature | Category | Height | Rainfall | High Humidity | canRain | SnowyBiome | enableSnow | classname | isMutation |   |   |   |  
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
openterraingenerator:alps | Alps | 0,00 | COLD | -1,80 | 0,10 | false | false | true | true | OTGBiome | false | FOREST | SNOWY | COLD | CONIFEROUS
biomesoplenty:alps | Alps | -0,25 | COLD | 7,82 | 0,30 | false | false | true | true | BiomeGenAlps | false | COLD | DRY | MOUNTAIN | SNOWY

**Comment by fllo** [comment]
Serene Seasons seems to be working without a problem with the fruittree spawing.

**Comment by ghost** [comment]
I know, I meant that in the context of not modifying biomes, but it was nullified by a previous comment.

And yeah, biomebundle is weird in ways.
